bandit/core/tester.py

```diff
@@ -32,13 +32,15 @@ def run_tests(self, raw_context, checktype):
 
         :param raw_context: Raw context dictionary
         :param checktype: The type of checks to run
-        :param nosec_lines: Lines which should be skipped because of nosec
-        :return: a score based on the number and type of test results
+        :return: a score based on the number and type of test results with
+                extra metrics about nosec comments
         '''
 
         scores = {
             'SEVERITY': [0] * len(constants.RANKING),
-            'CONFIDENCE': [0] * len(constants.RANKING)
+            'CONFIDENCE': [0] * len(constants.RANKING),
+            'nosecs_by_tests': 0,
+            'failed_nosecs_by_test': 0
```

Comment:
Why are we adding these here? We don't report `nosec` here from what I can see this file, we only use that to exclude those lines from checks

Reply:
I sort of shoehorned this in here because I couldn't think of any existing structures that were better suited for tracking these until we make it to metric collection. The trouble being that we check for test failures in this module, but we report the metrics separately and I needed a way to tie these results over to metrics so that they could be reported. This is complicated by the way that this method is called within BanditNodeVisitor, so here I'm pretty much calculating these metrics on the fly while visiting the AST.

Very open to ideas on changing this around.

Reply:
I ended up just passing metrics into `BanditTester`'s constructor when we instantiate it in `BanditNodeVisitor`'s constructor so we can just call the metric function there directly, is this okay? I originally did it this way to avoid changing any return types or function signatures.